ucs/database.cpp

```diff
@@ -280,16 +304,136 @@ bool Database::LoadChatChannels()
+	// Check if channelName is already in the BlockedChannelNames vector
+	bool isFound = false;
+	for (const std::string& name : BlockedChannelNames) {
+		std::string lowercaseName = name;
+		std::transform(lowercaseName.begin(), lowercaseName.end(), lowercaseName.begin(), ::tolower);
+		if (channelNameLower == lowercaseName) {
+			return true;
+		}
+	}
+	return false;
+}
+
+void Database::AddToChannelBlockList(std::string channelName) {
+	// Ignore the input if it is empty
+	if (channelName == "") {
+		return;
+	}
+	// Convert channelName to lowercase for case-insensitive comparison
+	std::string channelNameLower = channelName;
+	std::transform(channelNameLower.begin(), channelNameLower.end(), channelNameLower.begin(), ::tolower);
+	// Check if channelName is already in the BlockedChannelNames vector
+	bool isFound = false;
+	for (const std::string& name : BlockedChannelNames) {
+		std::string lowercaseName = name;
+		std::transform(lowercaseName.begin(), lowercaseName.end(), lowercaseName.begin(), ::tolower);
+		if (channelNameLower == lowercaseName) {
+			isFound = true;
+			break;
+		}
+	}
+	// Add channelName to the BlockedChannelNames vector if it is not already present
+	if (!isFound) {
+		BlockedChannelNames.push_back(channelName);
+	}
+}
+
+
+void Database::RemoveFromChannelBlockList(std::string channelName) {
+	if (!(channelName == "")) {
+		auto it = std::find(BlockedChannelNames.begin(), BlockedChannelNames.end(), channelName);
+		if (it != BlockedChannelNames.end()) {
+			BlockedChannelNames.erase(it);
+		}
+	}
+}
+
+bool Database::IsChatChannelInDB(std::string channelName)
+{
+	std::string rquery = fmt::format("SELECT COUNT(*) FROM `chatchannels` WHERE `name` = '{}'", channelName);
+	auto results = QueryDatabase(rquery);
+	auto row = results.begin();
+	int number_of_channels = std::stoul(row[0]);
+	if (number_of_channels > 0) {
+		LogDebug("Chat channel [{}] is in the database.", channelName);
+		return true;
+	}
+	LogDebug("Chat channel [{}] is not in the database.", channelName);
+	return false;
+}
+
+void Database::SaveChatChannel(std::string channelName, std::string channelOwner, std::string channelPassword, uint16 minstatus) {
+	if (IsOnChannelBlockList(channelName)) { //If channel name is blocked, do not save it to the database
+		return;
+	}
+	else if (this->IsChatChannelInDB(channelName)) { // If Channel exists, update it in the database
+		auto query = fmt::format("UPDATE chatchannels SET `name` = '{}', `owner` = '{}', `password` = '{}', `minstatus` = {}	WHERE name = '{}'", channelName, channelOwner, channelPassword, minstatus, channelName);
+		QueryDatabase(query);
+		LogDebug("Updating channel [{}], owned by [{}], in database.", channelName, channelOwner);
+		return;
+	}
+	else { // If channel does not exist, save it to the database
+		auto query = fmt::format("INSERT INTO chatchannels (`name`, `owner`, `password`, `minstatus`) VALUES('{}', '{}', '{}', {})", channelName, channelOwner, channelPassword, minstatus);
```

Comment:
Name should be escaped, someone could technically do SQL injection here

`Strings::Escape()`

Reply:
Updated